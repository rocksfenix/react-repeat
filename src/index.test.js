import * as React from 'react'
import Repeat from './index'
import renderer from 'react-test-renderer'

describe('Repeat', function () {
   it('should display one element', function () {
      const component = renderer.create(
         <Repeat size={1}>
           <li>üî•</li>
         </Repeat>,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
   })

   it('should repeat two elements twice', function () {
      const component = renderer.create(
         <Repeat size={4}>
           <li>üî•</li>
           <li>üçè</li>
         </Repeat>,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
   })

   it('should repeat a React Component six times', function () {
      const MyComp = function (props) {
         return (
            <button onClick={props.onClick}>Click</button>
         )
      }

      const component = renderer.create(
         <Repeat size={6}>
           <MyComp />
         </Repeat>,
      )

      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
   })

   it('should repeat a React Component and call listeners', function () {
      const MyComp = function (props) {
         return (
            <button onClick={props.onClick}>Click me!</button>
         )
      }

      const validateSpy = jest.fn()

      const component = renderer.create(
         <Repeat size={6}>
           <MyComp onClick={validateSpy} />
         </Repeat>
      )

      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      // manually trigger the callback
      tree[0].props.onClick();
      tree[0].props.onClick();
      expect(validateSpy).toHaveBeenCalledTimes(2)
   })

})
