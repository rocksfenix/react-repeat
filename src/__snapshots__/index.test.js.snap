// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repeat should display one element 1`] = `
<li
  index={1}
>
  🔥
</li>
`;

exports[`Repeat should repeat a React Component and call listeners 1`] = `
Array [
  <button
    onClick={[MockFunction]}
  >
    Click me!
  </button>,
  <button
    onClick={[MockFunction]}
  >
    Click me!
  </button>,
  <button
    onClick={[MockFunction]}
  >
    Click me!
  </button>,
  <button
    onClick={[MockFunction]}
  >
    Click me!
  </button>,
  <button
    onClick={[MockFunction]}
  >
    Click me!
  </button>,
  <button
    onClick={[MockFunction]}
  >
    Click me!
  </button>,
]
`;

exports[`Repeat should repeat a React Component six times 1`] = `
Array [
  <button>
    Click
  </button>,
  <button>
    Click
  </button>,
  <button>
    Click
  </button>,
  <button>
    Click
  </button>,
  <button>
    Click
  </button>,
  <button>
    Click
  </button>,
]
`;

exports[`Repeat should repeat two elements twice 1`] = `
Array [
  <li
    index={1}
  >
    🔥
  </li>,
  <li
    index={2}
  >
    🍏
  </li>,
  <li
    index={3}
  >
    🔥
  </li>,
  <li
    index={4}
  >
    🍏
  </li>,
]
`;
